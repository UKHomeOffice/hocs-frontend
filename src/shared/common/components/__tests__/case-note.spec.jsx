import React from 'react';
import CaseNote from '../case-note.jsx';

describe('When the component is rendered', () => {
    const mockDispatch = jest.fn();
    const useContextSpy = jest.spyOn(React, 'useContext').mockImplementation(() => ({
        dispatch: mockDispatch,
        page: {
            params: {
                caseId: '__caseId__'
            }
        },
        track: jest.fn(),
        hasRole: jest.fn().mockReturnValue(true)
    }));

    it('should match the snapshot', () => {
        const props = {
            author: '__author__',
            date: '__date__',
            modifiedBy: '__modifiedBy__',
            modifiedDate: '__modifiedDate__',
            note: '__note__',
            timelineItemUUID: '__timelineItemUUID__',
            title: '__title__'
        };
        const wrapper = shallow(<CaseNote {...props} />);

        expect(wrapper).toBeDefined();
        expect(useContextSpy).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });
});