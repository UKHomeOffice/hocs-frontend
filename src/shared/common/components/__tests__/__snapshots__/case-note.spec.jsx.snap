// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the component is rendered should match the snapshot 1`] = `
<Fragment>
  <p>
    <span
      className="case-note-number govuk-!-font-weight-bold"
    >
      __title__
      .
    </span>
    __note__
    <br />
  </p>
  <p>
    <span>
      Edited on __modifiedDate__ - __modifiedBy__
    </span>
  </p>
  <p>
    <span>
      __date__
    </span>
    <span>
      __author__
    </span>
    <span
      className="edit-link"
      onClick={[Function]}
    >
      <a
        href="#edit-case-note__timelineItemUUID__"
      >
        Edit
      </a>
    </span>
  </p>
</Fragment>
`;
